test/sbvr-helper.js

```diff
@@ -0,0 +1,763 @@
+			}
+			if (Array.isArray(lf)) {
+				se = toSE(lf, currentVocab);
+			} else if (_.isObject(lf)) {
+				({ lf, se } = lf);
+			}
+			this.se = getLineType(lf) + ': ' + se;
+			switch (lf[0]) {
+				case 'ConceptType':
+					term = lf[1];
+					const primitive = isPrimitive(term);
+					if (primitive) {
+						typeName = term[1];
+						const primitiveField = {
+							dataType: primitive,
+							fieldName: typeName,
+							required: true,
+							index: null,
+							references: null,
+							defaultValue: null,
+						};
+						Object.assign(this.table, {
+							idField: null,
+							primitive,
+							referenceScheme: typeName,
+							fields: [createdAtField, modifiedAtField, primitiveField],
+						});
+						this.matches = undefined;
+					} else {
+						typeName = term[1];
+						this.table.fields.push({
+							dataType: 'ConceptType',
+							fieldName: typeName,
+							required: true,
+							index: null,
+							references: {
+								fieldName: 'id',
+								resourceName: generateName(typeName),
+							},
+							defaultValue: null,
+						});
+						this.property = 'tables.' + this.table.resourceName;
+						this.matches = _.cloneDeep(this.table);
+					}
+					break;
+				case 'ReferenceScheme':
+					term = lf[1];
+					this.table.referenceScheme = term[1];
+					this.property = 'tables.' + this.table.resourceName;
+					this.matches = _.cloneDeep(this.table);
+					break;
+				case 'Necessity':
+					lf = nest(lf, [
+						'Rule',
+						'NecessityFormulation',
+						'UniversalQuantification',
+					]);
+					let quant = nest(lf, ['ExactQuantification']);
+					if (quant) {
+						card = nest(quant, ['Cardinality', 'Number']);
+					} else {
+						quant = nest(lf, ['AtMostNQuantification']);
+						card = nest(quant, ['MaximumCardinality', 'Number']);
+					}
+					if (card && card[1] === 1) {
+						const bindings = nest(
+							quant,
+							['AtomicFormulation', 'RoleBinding'],
+							true,
+						)[0];
+						const factType = nest(quant, ['AtomicFormulation', 'FactType']);
+						const primitiveTable = bindings
+							.map((binding) => getTable(binding[1][1]).table)
+							.find((table) => table.primitive);
+						baseTable = getTable(factType[1][1]).table;
+
+						if (baseTable === primitiveTable) {
+							// If the primitive table is also the base table then it's actually a unique
+							// constraint rather than a cardinality constraint
+							this.baseTable.fields.find(
+								(field) => field.fieldName === primitiveTable.name,
+							).index = 'UNIQUE';
+							this.matches = _.cloneDeep(this.baseTable);
+						} else {
+							if (primitiveTable) {
+								this.table = 'Attribute';
+							} else {
+								this.table = 'ForeignKey';
+							}
+							const dataType =
+								(primitiveTable != null
+									? primitiveTable.primitive
+									: undefined) != null
+									? primitiveTable != null
+										? primitiveTable.primitive
+										: undefined
+									: 'ForeignKey';
```

Comment:
Imo this looks hard to read:
```suggestion
							const dataType =
								(primitiveTable != null && primitiveTable.primitive != null)
									? primitiveTable.primitive
									: 'ForeignKey';
```